documents/transfers.md

```diff
@@ -0,0 +1,691 @@
+      {
+        "id": "trf_DSkl2lXWbiADZG",
+        "entity": "transfer",
+        "source": "order_DSkl2lBNvueOly",
+        "recipient": "acc_CNo3jSI8OkFJJJ",
+        "amount": 500,
+        "currency": "INR",
+        "amount_reversed": 0,
+        "notes": {
+          "branch": "Acme Corp Bangalore North",
+          "name": "Gaurav Kumar"
+        },
+        "fees": 2,
+        "tax": 0,
+        "on_hold": true,
+        "on_hold_until": 1670776632,
+        "recipient_settlement_id": null,
+        "created_at": 1570794714,
+        "linked_account_notes": [
+          "Acme Corp Bangalore North"
+        ],
+        "processed_at": 1570794772
+      }
+    ]
+  }
+}
+
+```
+-------------------------------------------------------------------------------------------------------
+
+### Fetch transfer
+
+```rb
+transferId = "trf_E7V62rAxJ3zYMo"
+
+Razorpay::Transfer.fetch(transferId)
+```
+
+**Parameters:**
+
+| Name          | Type        | Description                                 |
+|---------------|-------------|---------------------------------------------|
+| transferId*   | string      | The id of the transfer to be fetched  |
+
+**Response:**
+```json
+{
+  "id": "trf_E7V62rAxJ3zYMo",
+  "entity": "transfer",
+  "source": "pay_E6j30Iu1R7XbIG",
+  "recipient": "acc_CMaomTz4o0FOFz",
+  "amount": 100,
+  "currency": "INR",
+  "amount_reversed": 0,
+  "notes": [],
+  "fees": 1,
+  "tax": 0,
+  "on_hold": false,
+  "on_hold_until": null,
+  "recipient_settlement_id": null,
+  "created_at": 1579691505,
+  "linked_account_notes": [],
+  "processed_at": 1579691505
+}
+```
+-------------------------------------------------------------------------------------------------------
+
+### Fetch transfers for a settlement
+
+```rb
+recipientSettlementId = "setl_DHYJ3dRPqQkAgV"
+
+Razorpay::Transfer.all({
+   "recipient_settlement_id": recipientSettlementId
+})
+```
+
+**Parameters:**
+
+| Name          | Type        | Description                                 |
+|---------------|-------------|---------------------------------------------|
+| recipientSettlementId*   | string    | The recipient settlement id obtained from the settlement.processed webhook payload.  |
+
+**Response:**
+```json
+{
+  "entity": "collection",
+  "count": 1,
+  "items": [
+    {
+      "id": "trf_DGSTeXzBkEVh48",
+      "entity": "transfer",
+      "source": "pay_DGSRhvMbOqeCe7",
+      "recipient": "acc_CMaomTz4o0FOFz",
+      "amount": 500,
+      "currency": "INR",
+      "amount_reversed": 0,
+      "notes": [],
+      "fees": 2,
+      "tax": 0,
+      "on_hold": false,
+      "on_hold_until": null,
+      "recipient_settlement_id": "setl_DHYJ3dRPqQkAgV",
+      "created_at": 1568110256,
+      "linked_account_notes": [],
+      "processed_at": null
+    }
+  ]
+}
+```
+-------------------------------------------------------------------------------------------------------
+
+### Fetch settlement details
+
+```rb
+Razorpay::Transfer.fetch_settlements
+```
+
```

Comment:
`def self.fetch_settlements
      request.get "?expand[]=recipient_settlement"
 end`
 
 expand parameter already linked with url . This approach will not fail in the test case because regex is used